providers/implementations/kem/hybrid_kem.c

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "prov/hybrid_pkey.h"
+#include "hybrid_kem_local.h"
+
+#include <string.h>
+#include <openssl/evp.h>
+#include <openssl/core_names.h>
+#include <openssl/proverr.h>
+#include "crypto/evp.h"
+#include "prov/providercommon.h"
+#include "prov/provider_ctx.h"
+#include "internal/e_os.h"
+#include "internal/refcount.h"
+
+HYBRID_PKEY_CTX *ossl_hybrid_kem_newctx(void *provctx, unsigned int num_algs,
+                                        const char **names,
+                                        uint32_t *key_lengths)
+{
+    HYBRID_PKEY_CTX *ctx = OPENSSL_zalloc(sizeof(*ctx));
+
+    if (ctx == NULL)
+        return NULL;
+    ctx->num_algs = num_algs;
+    ctx->libctx = PROV_LIBCTX_OF(provctx);
+    ctx->propq = NULL;
+    ctx->names = names;
+    ctx->key_lengths = key_lengths;
+    return ctx;
+}
+
+void ossl_hybrid_kem_freectx(void *vctx)
+{
+    HYBRID_PKEY_CTX *ctx = (HYBRID_PKEY_CTX *)vctx;
+    unsigned int i;
+
+    if (ctx == NULL)
+        return;
+    for (i = 0; i < ctx->num_algs; i++)
+        EVP_PKEY_CTX_free(ctx->ctxs[i]);
```

Comment:
I don't see anywhere where you create these `EVP_PKEY_CTX`s. What did I miss?

Reply:
nothing.